logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiver.kt

```diff
@@ -317,13 +324,19 @@ class ConversationEventReceiverImpl(
                 )
             }
 
-    private suspend fun handleDeletedConversation(event: Event.Conversation.DeletedConversation) =
-        conversationRepository.deleteConversation(event.conversationId)
+    private suspend fun handleDeletedConversation(event: Event.Conversation.DeletedConversation): Either<CoreFailure, Unit> {
+        val conversation = conversationRepository.observeConversationDetailsById(event.conversationId).onlyRight().first()
+        return conversationRepository.deleteConversation(event.conversationId)
             .onFailure { coreFailure ->
                 kaliumLogger.withFeatureId(EVENT_RECEIVER).e("$TAG - Error deleting the contents of a conversation $coreFailure")
             }.onSuccess {
+                val senderUser = userRepository.observeUser(event.senderUserId).firstOrNull()
+                val deletedConversationNotification =
+                    localNotificationMessageMapper.fromDeletedConversationToLocalNotification(event, conversation.conversation, senderUser)
```

Comment:
Maybe lets move this mapping into the `EphemeralNotificationsManager`? 

I believe `ConversationEventReceiver` should not know about any notifications.
And it will add more sense to this manager: EventReceiver handles deleting the conversation and letting know `EphemeralNotificationsManager` that conversation was removed and `EphemeralNotificationsManager` receives that "signal", map it into the notification and passes to the UseCase.

Reply:
I follow the point, I manage to get something in between that will allow us later to expand this `EphemeralNotificationsManager` to other types of notifications, and removing the core mapping logic to localnotifications away from the `ConversationEventReceiver`